### ISSUE TITLE
[flaky test] TestEvaluations/testdata/native_histograms.test

### ISSUE THREAD

**Issue by bwplotka** [prometheus/prometheus#14411]
### What did you do?

Did PR but got unrelated test failure, looks like flaky https://github.com/prometheus/prometheus/actions/runs/9790656321/job/27032735386

![image](https://github.com/prometheus/prometheus/assets/6950331/7c8328ca-e331-4630-88d0-9b6281c703a0)

**Comment by krajorama** [comment]
Ran it >10000 times locally at 8849b7dcadc2960c70467734efce35d737be25f5, but nothing. Also from the logs it seems like the testrun was shorter than 5 minutes, so I don't really understand what could have gone wrong.

**Comment by bboreham** [comment]
I saw it again.
https://github.com/prometheus/prometheus/actions/runs/10080286169/job/27869658760

**Comment by bboreham** [comment]
And again https://github.com/prometheus/prometheus/actions/runs/10149193455/attempts/1

**Comment by krajorama** [comment]
Managed to reproduce locally:
```
$ go test -count 100 -timeout 300s -run ^TestEvaluations$ github.com/prometheus/prometheus/promql -tags=dedupelabels
--- FAIL: TestEvaluations (1.50s)
    --- FAIL: TestEvaluations/testdata/native_histograms.test (0.08s)
        test.go:115: 
            	Error Trace:	/home/krajo/go/github.com/krajorama/prometheus/promql/promqltest/test.go:115
            	            				/home/krajo/go/github.com/krajorama/prometheus/promql/promqltest/test.go:108
            	Error:      	Received unexpected error:
            	            	error in eval histogram_sum(sum(last_over_time(incr_sum_histogram[5m]))) (line 270) range mode: expected 30 for {} but got 20
            	Test:       	TestEvaluations/testdata/native_histograms.test
--- FAIL: TestEvaluations (1.59s)
    --- FAIL: TestEvaluations/testdata/native_histograms.test (0.07s)
        test.go:115: 
            	Error Trace:	/home/krajo/go/github.com/krajorama/prometheus/promql/promqltest/test.go:115
            	            				/home/krajo/go/github.com/krajorama/prometheus/promql/promqltest/test.go:108
            	Error:      	Received unexpected error:
            	            	error in eval histogram_sum(sum(last_over_time(incr_sum_histogram[5m]))) (line 270) range mode: expected 30 for {} but got 10
            	Test:       	TestEvaluations/testdata/native_histograms.test
--- FAIL: TestEvaluations (1.57s)
    --- FAIL: TestEvaluations/testdata/native_histograms.test (0.08s)
        test.go:115: 
            	Error Trace:	/home/krajo/go/github.com/krajorama/prometheus/promql/promqltest/test.go:115
            	            				/home/krajo/go/github.com/krajorama/prometheus/promql/promqltest/test.go:108
            	Error:      	Received unexpected error:
            	            	error in eval histogram_sum(sum(last_over_time(incr_sum_histogram[5m]))) (line 270) range mode: expected 30 for {} but got 10
            	Test:       	TestEvaluations/testdata/native_histograms.test
--- FAIL: TestEvaluations (1.87s)
    --- FAIL: TestEvaluations/testdata/native_histograms.test (0.08s)
        test.go:115: 
            	Error Trace:	/home/krajo/go/github.com/krajorama/prometheus/promql/promqltest/test.go:115
            	            				/home/krajo/go/github.com/krajorama/prometheus/promql/promqltest/test.go:108
            	Error:      	Received unexpected error:
            	            	error in eval histogram_sum(sum(last_over_time(incr_sum_histogram[5m]))) (line 270) range mode: expected 30 for {} but got 20
            	Test:       	TestEvaluations/testdata/native_histograms.test
--- FAIL: TestEvaluations (1.83s)
    --- FAIL: TestEvaluations/testdata/native_histograms.test (0.08s)
        test.go:115: 
            	Error Trace:	/home/krajo/go/github.com/krajorama/prometheus/promql/promqltest/test.go:115
            	            				/home/krajo/go/github.com/krajorama/prometheus/promql/promqltest/test.go:108
            	Error:      	Received unexpected error:
            	            	error in eval histogram_sum(sum(last_over_time(incr_sum_histogram[5m]))) (line 270) range mode: expected 30 for {} but got 10
            	Test:       	TestEvaluations/testdata/native_histograms.test
FAIL
FAIL	github.com/prometheus/prometheus/promql	180.699s
FAIL
```

Also did some analysis in https://cloud-native.slack.com/archives/C02KR205UMU/p1722332883894509?thread_ts=1722327916.939029&cid=C02KR205UMU
```
Ok so the input is like this:
Series 1: (t=0, sum=0) ,  (t=5, sum=1), ..,  (t=45, sum=9), (t=50, sum=10)
Series 2: (t=0, sum=0),   (t=5, sum=2), ..,  (t=45, sum=18), (t=50, sum=20)
So instant query at t=50 => 10+20 = 30 (edited) 
```

Based on the wrong result being 10 or 20, we're simply missing one of the series.

**Comment by krajorama** [comment]
Fixed in #14538
